### ISSUE TITLE
Invalid SELECT TYPE selector in allocate_with_source_22.f03

### ISSUE THREAD

**Issue by nncarlson** [nncarlson/gfortran.dg#6]
In the statement
```fortran
30:  select type (b(1))
```
The selector `b(1)` is not a named variable, and so requires an associate-name according to constraint C835 (F08); i.e.,
```fortran
  select type (b1 => b(1))
```
```
$ nagfor -kind=byte allocate_with_source_22.f03
NAG Fortran Compiler Release 6.1(Tozai) Build 6144
Error: allocate_with_source_22.f03, line 30: Syntax error
       detected at B@(
```

**Comment by janusw** [comment]
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=84385

**Comment by janusw** [comment]
Fixed with 6e701c39c421d65152d9c08d89b8abe9101a640d. Closing.
